### PR TITLE
Adds localization helper scripts

### PR DESCRIPTION
**Pull request by csabol-stripe**
## Summary

Adds new scripts to push new localized strings to lokalise, pull translations from lokalise, detect missing translations, and detect newly added localized strings.

CI will now include a check for any localized strings that were not added to en.lproj/Localizable.strings

## Motivation

Make it easier to keep track of our localizations

## Testing

Tested uploading and downloading from lokalise. Tested finding missing translations. Tested detecting new localized strings and removing one's that aren't used in code base.

### CODE REVIEW

**Comment by csabol-stripe** [comment]
I can't find documentation on lokalise's encoding, so I'm gonna hold off commiting this one until I get a response from them so I can determine if we need to use recode to convert to utf-8 or not

**Comment by csabol-stripe** [comment]
companion wiki page on when to use these coming ~tomorrow

File: ci_scripts/download_localized_strings_from_lokalise.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+if [[ -z $(which recode) ]]; then
+    echo "Installing recode via homebrew..."
+    brew install recode
+fi
+
+API_TOKEN=$(fetch-password mobile/lokalise/token -q)
+PROJECT_ID=$(fetch-password mobile/lokalise/ios -q)
+LANGUAGES="da,de,en-GB,en,es-AR,es,fi,fr-CA,fr,it,ja,ko,nb,nl,nn-NO,pt-BR,pt-PT,ru,sv,tr,zh-HANS,zh-HK,zh-TW"
+# Chinese (Traditional) is zh-TW by default in lokalise so we have to map it to zh-Hant
```
Comment (mkaplan9) [inline ci_scripts/download_localized_strings_from_lokalise.sh:17]: If you want you can change this in Lokalise

Click the flag, then `Language Settings`
image

Change `Custom language code` to zh-Hant
image

**Comment by csabol-stripe** [comment]
> All our exported files are encoded in UTF-8.

👍

File: ci_scripts/download_localized_strings_from_lokalise.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+lokalise2 --token $API_TOKEN \
+          --project-id $PROJECT_ID \
+          file download \
+          --format strings \
+          --filter-langs $LANGUAGES \
+          --language-mapping $LANG_MAP \
+          --export-sort "a_z" \
+          --directory-prefix "%LANG_ISO%.lproj" \
+          --original-filenames=true \
+          --unzip-to Stripe/Resources/Localizations/
```
Comment (mkaplan9) [inline ci_scripts/download_localized_strings_from_lokalise.sh:29]: The workflow we have with translators has than go through a few steps, and the string is only complete when it's in the `Final` state. To pull only `Final` keys you can add
```
--custom-translation-status-ids final_status_id
```

ios sdk final status id is 587
ios 3ds2 sdk final status id is 583

File: ci_scripts/upload_localized_strings_to_lokalise.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+fi
+
+API_TOKEN=$(fetch-password mobile/lokalise/token -q)
+PROJECT_ID=$(fetch-password mobile/lokalise/ios -q)
+
+lokalise2 --token $API_TOKEN \
+          --project-id $PROJECT_ID \
+          file upload \
+          --file Stripe/Resources/Localizations/en.lproj/Localizable.strings \
+          --lang-iso "en"
```
Comment (mkaplan9) [inline ci_scripts/upload_localized_strings_to_lokalise.sh:22]: Will the key name always == the string? If yes, this looks good, if not there are a few extra params you may want.

**Comment by mkaplan9** [comment]
> I can't find documentation on lokalise's encoding, so I'm gonna hold off commiting this one until I get a response from them so I can determine if we need to use recode to convert to utf-8 or not

Did you reach out to them yet? I not I can today

File: ci_scripts/download_localized_strings_from_lokalise.sh (hunk 2 above)
Comment (csabol-stripe) [reply ci_scripts/download_localized_strings_from_lokalise.sh:29]: oh cool, I'll add!

File: ci_scripts/upload_localized_strings_to_lokalise.sh (hunk 3 above)
Comment (csabol-stripe) [reply ci_scripts/upload_localized_strings_to_lokalise.sh:22]: yeah for the time being. I think in the future we'll probably want key,value,comment but right now we only support key/value, comment

File: ci_scripts/download_localized_strings_from_lokalise.sh (hunk 1 above)
Comment (csabol-stripe) [reply ci_scripts/download_localized_strings_from_lokalise.sh]: Yeah I like that better. Updated in lokalise

**Comment by mkaplan9** [comment]
good from i18n standpoint

**Pull request merged**